src/workerd/jsg/value.h

```diff
@@ -468,21 +469,22 @@ class StringWrapper {
       kj::Maybe<v8::Local<v8::Object>> parentObject) {
     // TODO(cleanup): Move to a HeaderStringWrapper in the api directory.
     v8::Local<v8::String> str = check(handle->ToString(context));
-    auto result =
-        ByteString(KJ_ASSERT_NONNULL(tryUnwrap(context, str, (kj::String*)nullptr, parentObject)));
-    if (str->ContainsOnlyOneByte()) {
-      auto buf = kj::heapArray<kj::byte>(str->Length() + 1);
-      str->WriteOneByte(context->GetIsolate(), buf.begin());
-      for (auto b: buf) {
-        if (b >= 128) {
-          result.warning = ByteString::Warning::CONTAINS_EXTENDED_ASCII;
-          break;
-        }
+    auto result = ByteString(KJ_ASSERT_NONNULL(
+        tryUnwrap(context, str, static_cast<kj::String*>(nullptr), parentObject)));
+
+    v8::String::ValueView str_view(context->GetIsolate(), str);
+
+    // This will avoid reading the whole underlying string.
+    if (str_view.is_one_byte()) [[likely]] {
+      if (!simdutf::validate_ascii(
+              reinterpret_cast<const char*>(str_view.data8()), str_view.length())) [[unlikely]] {
+        result.warning = ByteString::Warning::CONTAINS_EXTENDED_ASCII;
       }
-    } else {
-      result.warning = ByteString::Warning::CONTAINS_UNICODE;
+      return kj::mv(result);
```

Comment:
I wonder if having two return statements -- particularly one with `kj::mv` and one without -- will confuse the compiler out of return value optimization? I would change this back to using an `else` block and have a single `return result;` at the end.

Reply:
Agreed. :+1: